src/components/ContextMenu.tsx

```diff
@@ -0,0 +1,81 @@
+/*!
+ * © 2024 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement, useMemo, useState } from 'react'
+import styled from 'styled-components'
+
+import { IconButton, IconButtonGroup } from './Button'
+import { AddComment, EditIcon } from './icons'
+import { Tooltip } from './Tooltip'
+
+export interface Actions {
+  label: string
+  action: () => void
+  icon: string
+}
+
+export interface ContextMenuProps {
+  actions: Actions[]
+}
+
+const ContextMenuIconButton = styled(IconButton)`
+  color: #6e6e6e;
+  &:not([disabled]):hover,
+  &:not([disabled]):focus {
+    color: #363636;
+    background-color: #f2f2f2;
+    border-color: #f2f2f2;
+  }
+`
+
+export const ContextMenu: React.FC<ContextMenuProps> = ({ actions }) => {
+  // IconSelector can be expanded with other icons/actions
+  const IconSelector: React.FC<{ iconName: string }> = ({ iconName }) => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const icons: { [key: string]: React.FC<any> } = {
```

Comment:
why do you need to specify the type here? will it not be implicitly derived out of the content of the object?

Reply:
without types the typecheck returns errors:

src/components/ContextMenu.tsx:58:27 - error TS7031: Binding element 'iconName' implicitly has an 'any' type.
58   const IconSelector = ({ iconName }) => {
                             ~~~~~~~~
src/components/ContextMenu.tsx:64:28 - error TS7053: Element implicitly has an 'any' type because expression of type 'any' can't be used to index type '{ AddComment: () => Element; EditIcon: FC<SVGAttributes<SVGElement>>; }'.
64     const Icon: React.FC = icons[iconName]
                              ~~~~~~~~~~~~~~~
Found 2 errors in the same file, starting at: src/components/ContextMenu.tsx:58

Reply:
try to replace `iconName: string` to `iconName: 'AddComment' | 'EditIcon'` and the Actions type:
```
export interface Actions {
  label: string
  action: () => void
  icon: 'AddComment' | 'EditIcon'
}
```

with that it works for me without specifying the type for the icons object

Reply:
There can be more icons in future...